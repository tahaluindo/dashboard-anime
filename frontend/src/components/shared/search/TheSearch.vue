<template>
  <span>
    <form-field
      v-model:data-from-parent="search"
      :lazy="true"
      labelName="pesquisar"
    />
  </span>
</template>

<script lang="ts">
import { defineComponent, nextTick, ref, watchEffect } from "vue";
import FormField from "../forms/FormField.vue";

export default defineComponent({
  components: { FormField },
  emits: ["onSearch"],
  setup(_, { emit }) {
    const search = ref("");

    watchEffect(async () => {
      if (search.value) {
        emit("onSearch", search.value);
        await nextTick();
        search.value = "";
      }
    });

    return {
      search,
    };
  },
});
</script>

<style lang="postcss"></style>
