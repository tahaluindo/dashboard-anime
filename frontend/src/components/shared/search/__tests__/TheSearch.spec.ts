import { mount } from "@vue/test-utils";
import TheSearch from "../TheSearch.vue";

describe("TheSearch component test.", () => {
  it("should render the default component.", async () => {
    const wrapper = mount(TheSearch);

    expect(wrapper.emitted()).not.toContain("onSearch");

    await wrapper.find("input").setValue("test");

    const emitted: Array<Array<string>> = wrapper.emitted("onSearch") || [[]];
    expect(emitted[0][0]).toBe("test");

    expect(wrapper.html()).toMatchSnapshot();
  });
});
