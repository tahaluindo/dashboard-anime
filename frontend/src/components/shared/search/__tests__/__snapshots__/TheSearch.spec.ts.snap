// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSearch component test. should render the default component. 1`] = `"<span><div class=\\"relative h-16 w-full\\"><input class=\\"bg-none p-3 pb-1 w-full border-b-2 border-t-0 border-r-0 border-l-2 border-current text-current rounded focus:border-secondary focus:ring-0\\" type=\\"text\\" name=\\"pesquisar\\" id=\\"pesquisar\\"><label class=\\"absolute rounded top-1 on-focus left-3 text-current bg-none z-10 p-1 duration-450\\" for=\\"pesquisar\\">pesquisar</label></div></span>"`;
